
Get the tempalte friends to compile with gcc-3.4.

  - Rene Rebe <rene@exactcode.de

--- ./lib/list.h.vanilla	2004-07-05 13:54:18.000000000 +0200
+++ ./lib/list.h	2004-07-05 14:01:49.000000000 +0200
@@ -9,18 +9,19 @@
   ~list_node() { }
 };
 
-template<class T> class list_iterator;
-template<class T> class const_list_iterator;
+template<typename T> class list_iterator;
+template<typename T> class const_list_iterator;
 
-template<class T> class list
+template<typename T> class list
 {
 public:
   typedef list_node<T> node;
   typedef list_iterator<T> iter;
   typedef const_list_iterator<T> const_iter;
-  friend class iter;
-  friend class const_iter;
-  
+
+  friend class list_iterator<T>;
+  friend class const_list_iterator<T>;
+ 
   list()
     : head(0), tail(0), cnt(0)
     {
